libraries/node-core-library/src/LockFile.ts

```diff
@@ -6,28 +6,105 @@ import * as path from 'path';
 import * as child_process from 'child_process';
 import { setTimeout } from 'timers';
 
+/**
+ * Parses the process start time from a linux /proc/1/stat file.
+ * @param stat The contents of a linux /proc/1/stat file.
+ * @returns The process start time in yiffies.
+ */
+export function getProcessStartTimeFromProcStat (stat: string): string | undefined {
+  // Parse value 22.
+  // We cannot just split stat on spaces, because value 2 may contain spaces.
+  // For example, when running the following Shell commands:
+  // > cp "$(which bash)" ./'bash 2)('
+  // > ./'bash 2)(' -c 'OWNPID=$BASHPID;cat /proc/$OWNPID/stat'
+  // 59389 (bash 2)() S 59358 59389 59358 34818 59389 4202496 329 0 0 0 0 0 0 0 20 0 1 0
+  // > rm -rf ./'bash 2)('
+  // The output shows a stat file such that value 2 contains spaces.
+  // To still umambiguously parse such output we assume all values after the third consist of only digits...
+
+  // trimRight to remove the trailing line terminator.
+  let values: string[] = stat.trimRight().split(' ');
+  let i: number = values.length - 1;
+  while (i >= 0 && /^[0-9]+$/.test(values[i])) {
+    i -= 1;
+  }
+  // i is the index of the third value (but i need not be 2).
+  if (i < 2) {
+    // Format of stat has changed.
+    return undefined;
+  }
+  const value2: string = values.slice(1, i - 1).join(' ');
+  values = [values[0], value2].concat(values.slice(i));
+  if (values.length < 22) {
+    // Older version of linux, or non-standard configuration of linux.
+    return undefined;
+  }
+  const startTimeYiffies: string = values[21];
+  // In theory, the representations of start time returned by /proc/*/stat and ps -o lstart can change while the
+  // system is running, but we assume this does not happen.
+  // So the caller can safely use this value as part of a unique process id (on the machine, without comparing
+  // accross reboots).
+  return startTimeYiffies;
+}
+
 /**
  * Helper function that is exported for unit tests only.
  * Returns undefined if the process doesn't exist with that pid.
  */
 export function getProcessStartTime(pid: number): string | undefined {
+  // Use toFixed() to ensure decimal representation is used when converting the PID to string, although no PID is
+  // large enough to be printed using exponential notation.
```

Comment:
>using exponential notation [](start = 32, length = 26)

FYI `1e+31.toFixed().toString() === '1e+31'`

Reply:
Wow! You're right!